pkgs/applications/science/biology/neuron/default.nix

```
@@ -1,94 +1,103 @@
 { lib, stdenv
 , fetchurl
-, pkg-config
-, automake
-, autoconf
-, libtool
-, ncurses
 , readline
-, which
-, python ? null
-, useMpi ? false
+, xorg
 , mpi
-, iv
+, cmake
+, bison
+, flex
+, git
+, perl
+, gsl
+, xcbuild
+, python3
+, usempi ? false
+, useiv ? true
+, core ? false
+, rx3d ? false
 }:
 
 stdenv.mkDerivation rec {
-  pname = "neuron${lib.optionalString useMpi "-mpi"}";
-  version = "7.5";
-
-  nativeBuildInputs = [ which pkg-config automake autoconf libtool ];
-  buildInputs = [ ncurses readline python iv ]
-    ++ lib.optional useMpi mpi;
-
+  pname = "neuron";
+  version = "8.2.1";
+  # format is for pythonModule conversion
+  format = "other";
+  propagatedBuildInputs = [
+    readline
+  ] ++ lib.optional useiv [
+    xorg.libX11
+    xorg.libX11.dev
+    xorg.libXcomposite
+    xorg.libXcomposite.dev
+    xorg.libXext
+    xorg.libXext.dev
+  ] ++ lib.optional usempi [
+    mpi
+  ];
+  nativeBuildInputs = [
+    cmake
+    bison
+    flex
+    git
+    python3
+    python3.pkgs.wheel
+    python3.pkgs.setuptools
+    python3.pkgs.scikit-build
+    python3.pkgs.matplotlib
+  ] ++ lib.optional core [
+    perl
+    gsl
+  ]
+  ++ lib.optional usempi [
+    python3.pkgs.mpi4py
+  ]
+  ++ lib.optional rx3d [
+    python3.pkgs.cython
+    python3.pkgs.numpy
+  ]
+  ++ lib.optional stdenv.isDarwin [
+    xcbuild
+  ];
+  patches = [
+    ./neuron_darwin_rpath.patch
+  ];
+  # Patch build shells for cmake (bin, src, cmake) and submodules (external)
+  postPatch = ''
+          patchShebangs ./bin ./src ./external ./cmake
+          sed -e 's#DESTDIR =#DESTDIR = '"$out"'#' -i external/coreneuron/extra/nrnivmodl_core_makefile.in
+        '';
+  cmakeFlags = [
```

```suggestion
  ];

  patches = [
    ./neuron_darwin_rpath.patch
  ];

  # Patch build shells for cmake (bin, src, cmake) and submodules (external)
  postPatch = ''
    patchShebangs ./bin ./src ./external ./cmake
    sed -e 's#DESTDIR =#DESTDIR = '"$out"'#' -i external/coreneuron/extra/nrnivmodl_core_makefile.in
  '';

  cmakeFlags = [
```
putting some whitespace between the attributes makes the code way more readable.